boltzrpc/boltzrpc.proto

```diff
@@ -497,6 +530,58 @@ message CreateReverseSwapResponse {
     optional string invoice = 5;
 }
 
+message CreateChainSwapRequest {
+    // amount of satoshis to swap
+    uint64 amount = 1;
+    Pair pair = 2;
+    // address where funds will go if the swap succeeds
+    optional string claim_address = 3;
+    // address where the coins should go if the swap fails.
+    optional string refund_address = 4;
+    // Wallet from which the swap should be paid from. Ignored if `external_pay` is set to true.
+    // If the swap fails, funds will be refunded to this wallet aswell.
+    optional string from_wallet = 5;
+    // wallet where the the funds will go if the swap succeeds.
+    optional string to_wallet = 6;
+    // Whether the daemon should broadcast the claim transaction immediately after the lockup transaction is in the mempool.
+    // Should only be used for smaller amounts as it involves trust in boltz.
+    optional bool accept_zero_conf = 7;
+    // If set, the daemon will not pay the swap from an internal wallet.
+    optional bool external_pay = 8;
+}
+
+message ChainSwapInfo {
+    string id = 1;
+    Pair pair = 2;
+    SwapState state = 3;
+    string error = 4;
+    string status = 5;
+    string preimage = 6;
+    bool is_auto = 8;
+    optional uint64 service_fee = 9;
+    float service_fee_percent = 10;
+    optional uint64 onchain_fee = 11;
+    int64 created_at = 12;
+    string entity = 13;
+    ChainSwapData from_data = 14;
+    ChainSwapData to_data = 15;
+}
+
+message ChainSwapData {
+    string id = 1;
+    Currency currency = 2;
+    string private_key = 3;
+    string their_public_key = 4;
+    int64 amount = 6;
+    uint32 timeout_block_height = 7;
+    string lockup_transaction_id = 8;
+    string transaction_id = 9;
+    string wallet = 11;
+    string address = 12;
```

Comment:
Is it possible that there is no address yet when a wallet and no address is set in the request to create?

Reply:
Yes. Should we make those optional?

Reply:
Yes